### DISCUSSION TITLE
[c-sharp-operations] custom Query root type is not supported

### DISCUSSION THREAD

**Discussion by joshlrogers** [dotansimha/graphql-code-generator#8393] [Q&A] [answered]
I am fairly new to GraphQL in general, but I am having the hardest time getting queries written in a way that the c-sharp-operations code generator will work.  Even on the most basic of queries that work when querying the endpoint but won't work for codegen:
```
query getWebhookSubscription($id: ID!) {    
    webhookSubscription(id: $id) {
        id
        topic
    }
}
```
This query results in an error during codegen:
```
[STARTED] Generate to ./Operations/GraphQLOperations.cs
[STARTED] Load GraphQL schemas
[SUCCESS] Load GraphQL schemas
[STARTED] Load GraphQL documents
[SUCCESS] Load GraphQL documents
[STARTED] Generate
[FAILED] Cannot read properties of undefined (reading 'fields')
```

My configuration file looks like the following:
```
import { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
  debug: true,
  verbose: true,
  schema: [
        {
            'https://www.mygraphqlurl.net/api/graphql.json': {
                'headers': {
                    'Access-Token': 'Token'
                }
            }
        }
    ],
    generates: {      
      './Operations/GraphQLOperations.cs': {
          plugins: [ 'c-sharp-operations', 'c-sharp' ],
          documents: './Operations/**/*.graphql',  
          overwrite: true,
          config: {
              namespaceName: 'Client.GraphQL',
              jsonAttributesSource: 'System.Text.Json',
              typesafeOperation: true,
              namingConvention: {
                  'typeNames': 'change-case-all#pascalCase',
                  'enumValues': 'change-case-all#upperCase',
                  'transformUnderscore': true
              },
              scalars: {
                  'ARN': 'string',
                  'ARN!': 'string',
                  'DateTime!': 'DateTimeOffset',
                  'DateTime': 'DateTimeOffset?',
                  'Decimal': 'Decimal?',
                  'Decimal!': 'Decimal',
                  'JSON': 'string',
                  'JSON!': 'string',
                  'Money': 'Decimal?',
                  'Money!': 'Decimal',
                  'UnsignedInt64': 'UInt64?',
                  'UnsignedInt64!': 'UInt64',
                  'URL': 'string',
                  'URL!': 'string',
                  'UtcOffset': 'DateTimeOffset?',
                  'UtcOffset!': 'DateTimeOffset?'
              }
          }
      },
      './graphql.schema.json': {
          plugins: [ 'introspection' ]
      }
  } 
};

export default config
```

I have tried using a fragment instead to represent the fields, but I get the same error.  As soon as I delete the query though the other graphql mutations that I have get generated and they work as expected. This also works if I turn off `typesafeOperations`. Could someone please point me in the right direction, this has been blocking me from any progress for almost a day now.

**Comment by joshlrogers** [comment]
It seems this has to do more with how my schema is being built than anything else. If not using `type Query` then this error appears.  Unfortunately, the schema I am working with is using `type QueryRoot`.  Here is a reproducible example:
![image](https://user-images.githubusercontent.com/6315573/192607035-88a6425a-269b-4a1b-b27c-4a96ff9e4037.png)
And here is the same thing working but with `type Query`
![image](https://user-images.githubusercontent.com/6315573/192607373-addcf48a-a0cf-40e3-86b8-cc26baceb61f.png)

**Comment by charlypoly** [comment, marked as answer]
Hi @joshlrogers,

It seems like this plugin does not support setting a custom Query root type for now.
I'll add it to my todolist.

In the meantime, if you are willing to take a look, feel free to start a PR; I'll be happy to help you make it move forward 👀
